test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+ * Content-Type: image/png
+ *
+ * <binary data>
+ * ----------------------------732677279853170760492713--
+ * ```
+ *
+ * @param {THttpClient} httpClient
+ * @param {string} resourceUrl is the url of the resource for which you want to put an attachment
+ *                              for example https://localhost:5000/partiesS3/<some-guid>
+ * @param {Array<TFileToUpload | TFileToCopy>} fileDetails
+ * @returns {Promise<THttpResponse>} a http response
+ */
+async function doPutFiles(httpClient, resourceUrl, fileDetails) {
+  const body = createUploadBody(fileDetails);
+  const formData = new FormData();
+  formData.append('body', JSON.stringify(body));
+
+  fileDetails
+    .forEach((f) => {
+      if ('localFilename' in f) {
+        const { remotefileName, localFilename, attachmentKey } = f;
+        formData.append(attachmentKey, fs.createReadStream(localFilename), {
+          filename: remotefileName,
+          contentType: 'image/png',
+        })
+      }
+    });
+
+  // Currently in sri4node a streaming request automatically results in a streaming response.
+  // -> process the response streaming to be able to catch errors (see getStreamAsBuffer function
+  //    for more information).
+  const response = await httpClient.post({
+    path: resourceUrl + "/attachments",
+    headers: formData.getHeaders(), // needed to make it a multipart request !
+    body: formData,
+    streaming: true, // ask for streaming response
+  });
+
+  const responseBody = (await getStreamAsBuffer(response.body)).toString();
+  return {
+    ...response,
+    body: responseBody,
+  }
+}
+
+/**
+ * Reads a stream into a buffer.
+ * 
+ * In case of errors during a streaming response (at this point the client already received a http 200 with headers),
+ * the only way a server can indicate this is to drop the connection (and eventually dump an error message as text 
+ * into the stream). To be able to check error messages in failure test cases, the partial body which has been read
+ * before the error and which might contain an error message is thrown to the client in a wrapped error object.
+ *
+ * @param {TReadableStream} stream
+ * @returns {Promise<Buffer>}
+ */
+async function getStreamAsBuffer(stream) {
+  const chunks = [];
+  try {
+    for await (const chunk of stream) {
+      chunks.push(chunk);
+    }
+    return Buffer.concat(chunks);
+  } catch (err) {
+    throw {
+      code: 'streaming.body.read.failed',
+      err,
+      partialBody: Buffer.concat(chunks),
+    };
+  }
+}
+
+/**
+ * Will throw an assertion error if the two streams are not equal.
+ *
+ * We'll collect all bytes first because we only expect short files!
+ *
+ * @param {TReadableStream} s1
+ * @param {TReadableStream} s2
+ * @throws {AssertionError}
+ */
+async function checkStreamEqual(s1, s2) {
+  const b1 = await getStreamAsBuffer(s1);
+  const b2 = await getStreamAsBuffer(s2);
+  if (b1.byteLength !== b2.byteLength) {
```

Comment:
Why not simply use https://nodejs.org/docs/latest-v16.x/api/buffer.html#bufequalsotherbuffer?